vignettes/datatable-programming.Rmd

```diff
@@ -121,6 +121,47 @@ There are third party packages that can achieve what base R computing on the lan
 
 Though these can be helpful, we will be discussing a `data.table`-unique approach here.
 
+### Dynamic Grouping and Naming Syntax
+
+Besides the programmatic use with `env`, data.table offers some powerful and concise syntax for interactive use, especially in the `by` argument.
+
+```{r}
+d = data.table(x = 1:4, y = 2:5)
+```
+
+#### Grouping by Expressions in `by`
+
+For convenience, data.table allows you to group by a single expression directly without `list()` or `.()`. To name the resulting grouping column, you have two options:
+
+```{r}
+# 1. The canonical way: a named list (required for multiple expressions)
+d[, sum(y), by = .(grp = x %% 2)]
+
+# 2. A concise shortcut: parentheses (for a single expression)
```

Comment:
Saving single character is not worth advertising, especially because it was technically needed as a workaround AFAIR.
Moreover I don't see the reason of editing this file at all!
Metaprogramming is about specifying variables or functions by their names stored in a string. This is not the case here...